### PR TITLE
DRA API: validate node selector labels

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Previously, ValidateNodeSelector did not check that labels are valid. Now it does for resource.k8s.io, regardless whether an object already was created with invalid labels in an earlier Kubernetes release. Theoretically this is a breaking change and could cause problems during an upgrade, but that is highly unlikely in practice.

In contrast to node affinity, DRA does not ignore parse errors (= uses NewNodeSelector, not NewLazyErrorNodeSelector), so invalid labels would have been found instead of being silently ignored.

Even if some object has invalid labels, this only affects an alpha -> beta upgrade which isn't guaranteed to work seamlessly.

#### Which issue(s) this PR fixes:

Related-to: https://github.com/kubernetes/kubernetes/issues/128156

#### Special notes for your reviewer:

This PR is a subset of https://github.com/kubernetes/kubernetes/pull/128212. It was stripped down to just the parts which strengthen validation of resource.k8s.io.

We need to get this included in 1.32 because the resource.k8s.io API gets promoted to beta in 1.32. Making this change later would be more complicated.

#### Does this PR introduce a user-facing change?
```release-note
DRA: labels in node selectors now are validated. Invalid labels already caused runtime errors before and are unlikely to occur in practice.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/4381
```

/cc @AxeZhan @thockin @liggitt @kubernetes/release-team-leads

### CODE REVIEW

**Comment by liggitt** [comment]
change lgtm
/lgtm
/approve

I'm strongly in favor of tightening the validation prior to the beta API release. @thockin, would you agree?

If so, this will need milestoning by the release team, ideally ahead of the rc.0

**Comment by pohly** [comment]
/hold

For comment by @thockin, in case that we get the milestone first.

**Review by fsmunoz — APPROVED** [review]

**Comment by thockin** [comment]
ACK to fix thise before 32.

/approve
/remove hold

**Pull request merged**
